# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxMacPlatformFontList.h
# Commit: 0d746825f92a
# Full Hash: 0d746825f92a3039433aa275997c6e7f2ebfaf56
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-18 21:36:28
# Regressor Bug: 1708768
# File Overlap Count: 2
# Description:
#   Bug 1708768 - Make InitFontList safe to call off-main-thread on macOS, and run font-list initialization on a separate thread during startup. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115295
# ==============================================================================

diff -r fbe7310ee5f3 -r 0d746825f92a gfx/thebes/gfxMacPlatformFontList.h
--- a/gfx/thebes/gfxMacPlatformFontList.h	Tue May 18 11:10:43 2021 +0000
+++ b/gfx/thebes/gfxMacPlatformFontList.h	Tue May 18 11:10:43 2021 +0000
@@ -175,6 +175,10 @@
   nsresult InitFontListForPlatform() override;
   void InitSharedFontListForPlatform() override;
 
+  // handle commonly used fonts for which the name table should be loaded at
+  // startup
+  void PreloadNamesList();
+
   // special case font faces treated as font families (set via prefs)
   void InitSingleFaceList();
   void InitAliasesForSingleFaceList();
@@ -235,6 +239,9 @@
   bool mUseSizeSensitiveSystemFont;
   nsCString mSystemTextFontFamilyName;
   nsCString mSystemDisplayFontFamilyName;  // only used on OSX 10.11
+
+  nsTArray<nsCString> mSingleFaceFonts;
+  nsTArray<nsCString> mPreloadFonts;
 };
 
 #endif /* gfxMacPlatformFontList_H_ */